src/features/wallet/helpers/collectRewards.ts

```diff
@@ -0,0 +1,76 @@
+import { Dispatch, SetStateAction } from "react";
+import { ethers } from "ethers";
+import { RewardsRetrieved } from "web3/get/queryRewardsRetrievedEvents";
+import {
+  PostRetrieveReward,
+  PendingRequestRetrieveReward,
+} from "web3/post/retrieveRewards";
+
+import VotingArtifact from "@uma/core/build/contracts/Voting.json";
+
+const DEFAULT_BALANCE = "0";
+
+interface MulticallCollectRewards {
+  // contract address
+  target: string;
+  callData: string;
+}
+
+export default function collectRewards(
+  contract: ethers.Contract,
+  data: RewardsRetrieved[],
+  // Type returned by useState variable in Wallet.tsx
+  setAvailableRewards: Dispatch<SetStateAction<string>>,
+  multicallContract: ethers.Contract
+) {
+  const multicallCollectRewards = [] as MulticallCollectRewards[];
+  const votingInterface = new ethers.utils.Interface(VotingArtifact.abi);
+
+  // find the unique roundIds
+  const uniqueRoundIds = data
+    .map((item) => item.roundId)
+    .filter((value, index, self) => self.indexOf(value) === index);
```

Comment:
I really wish that the array had a native `.unique()` method rather than having to implement it yourself or use lodash, so this wouldn't be required to do something so simple.